composer.json

```diff
@@ -28,7 +28,7 @@
     "phpspec/prophecy": "^1.10.3",
     "phpunit/phpunit": "^9.1",
     "prooph/php-cs-fixer-config": "^0.3.1",
-    "vimeo/psalm": "^3.11.2"
+    "vimeo/psalm": "^3.11.2,<3.12"
```

Comment:
why this `<3.12` ?

Reply:
Because later versions adds psalm warnings which fails the build. (see log here: [https://travis-ci.com/github/prooph/event-store/jobs/461215042](https://travis-ci.com/github/prooph/event-store/jobs/461215042)) 

My suggestion is to fix this in another PR. If you disagree though, we need to make the psalm checks successful. I havn't found any other solution than removing all the affected @psalm-pure annotations.

Reply:
Found this: https://github.com/vimeo/psalm/issues/4114

Instead of @psalm-pure, use @psalm-mutation-free 

I can change this throughout the code (more than 100 places I think) and fix some other psalm related stuff. Just let me know.